### PR TITLE
Add status missing to update snapshot

### LINKED ISSUE #6142: Status MISSING no longer present in update report
In the update report, there used to be a status called `MISSING`, which was when there was no response for an observation. This no longer exists, and we only have `INACTIVE`. This should be reserved for observations where it is deactivated because of outlier detection.

### PR DESCRIPTION
**Pull request by oyvindeide**
**Issue**
Resolves #6142

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

## Pre review checklist

- [x] Read through the code changes carefully after finishing work
- [x] Make sure tests pass locally (after every commit!)
- [x] Prepare changes in small commits for more convenient review (optional)
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Updated documentation
- [x] Ensured that unit tests are added for all new behavior (See 
    [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)),
    and changes to existing code have good test coverage.

## Pre merge checklist
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/ert/analysis/_es_update.py [hunk 1]
```diff
@@ -49,16 +49,24 @@ class ErtAnalysisError(Exception):
...
-    obs_mask: npt.NDArray[np.bool_]
-    response_mean: npt.NDArray[np.float32]
-    response_std: npt.NDArray[np.float32]
-
-    @property
-    def obs_status(self) -> List[str]:
-        return ["ACTIVE" if v else "DEACTIVATED" for v in self.obs_mask]
+    name: str
+    value: float
+    std: float
```
Comment (dafeda) [inline src/ert/analysis/_es_update.py:54]: Why rename from `obs_std` to `std`?

File: src/ert/analysis/_es_update.py [hunk 2]
```diff
@@ -347,17 +355,32 @@ def _load_observations_and_responses(
...
-    update_snapshot = UpdateSnapshot(
-        obs_name=obs_keys,
-        obs_value=observations,
-        obs_std=errors,
-        obs_mask=obs_mask,
-        response_mean=ens_mean,
-        response_std=ens_std,
-    )
+    update_snapshot = []
+    for name, obs_val, obs_std, status, mean, std, mean_mask, std_mask in zip(
```
Comment (dafeda) [inline src/ert/analysis/_es_update.py:361]: Are these name-changes necessary? I find them a bit confusing.

File: src/ert/analysis/_es_update.py (hunk 1 above)
Comment (oyvindeide) [reply src/ert/analysis/_es_update.py:54]: I removed the prefix as I figured it was more centered around observations. Could rename from `UpdateSnapshot` to `ObservationSnapshot` instead perhaps?

File: src/ert/analysis/_es_update.py (hunk 2 above)
Comment (oyvindeide) [reply src/ert/analysis/_es_update.py:361]: Compared with the class? Can align it to them. Not very fond of this huge zip, am happy if you have alternatives.

File: src/ert/analysis/_es_update.py (hunk 1 above)
Comment (dafeda) [reply src/ert/analysis/_es_update.py:54]: It's really `ObservationAndResponseSnapshot` 😬

File: src/ert/analysis/_es_update.py [hunk 3]
```diff
@@ -347,17 +355,32 @@ def _load_observations_and_responses(
...
+        obs_keys,
+        observations,
+        errors,
```
Comment (dafeda) [inline src/ert/analysis/_es_update.py:364]: `errors` is not used.

**Review by dafeda — APPROVED** [review]

**Pull request merged**